### PR TITLE
Feature/plot purge

### PR DESCRIPTION
**Pull request by Jammy2211**
When I put visualization in autofti, I put a lot of extra crap in I have now purged.

Along the way, I restructued all searches to spread their classes across Python packages.

### CODE REVIEW

File: autofit/plot/output.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            The format of the output, 'show' displays on the computer screen, 'png' outputs to .png, 'fits' outputs to
+            `.fits` format.
+        bypass
+            Whether to bypass the `plt.show` or `plt.savefig` methods, used when plotting a subplot.
+        """
+        self.path = path
+
+        if path is not None and path:
+            try:
+                os.makedirs(path)
```
Comment (rhayes777) [inline autofit/plot/output.py:62]:
```python
os.makedirs(
    path,
    exists_ok=True
)

File: autofit/plot/output.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            except FileExistsError:
+                pass
+
+        self.filename = filename
+        self._format = format
+        self.bypass = bypass
+
+    @property
+    def format(self) -> str:
+        if self._format is None:
```
Comment (rhayes777) [inline autofit/plot/output.py:72]: return self._format or None

**Review by rhayes777 — APPROVED** [review]

File: autofit/non_linear/mcmc/emcee/samples.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            The total number of steps taken by each walker of this MCMC `NonLinearSearch` (the total samples is equal
+            to the total steps * total walkers).
+        """
+
+        self.backend = backend
+
+        parameter_lists = self.backend.get_chain(flat=True).tolist()
+
+        log_prior_list = [
+            sum(model.log_prior_list_from_vector(vector=vector)) for vector in parameter_lists
```
Comment (rhayes777) [inline autofit/non_linear/mcmc/emcee/samples.py:41]: All this stuff _definitely_ needs to be in the super constructor. Refactoring is very difficult when the same code is repeated 10 times

**Review by rhayes777 — APPROVED** [review]

File: autofit/non_linear/mcmc/emcee/samples.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    @property
+    def total_walkers(self):
+        return len(self.backend.get_chain()[0, :, 0])
+
+    @property
+    def total_steps(self):
+        return len(self.backend.get_log_prob())
+
+    @property
+    def auto_correlations(self):
```
[the remark this answers is not in the document]
Comment (Jammy2211) [reply autofit/non_linear/mcmc/emcee/samples.py:90]: Will do in the future, wasnt the aim of this PR and not got much time atm.

**Pull request merged**